### ISSUE TITLE
FR: Support for LZ4?

### ISSUE THREAD

**Issue by gcflymoto** [pycompression/xopen#163]
First of all thank you for such an excellent library which provides our engineering community with simple, transparent, performant, and pythonic library to handle all kinds of compressed and uncompressed files. It reduces the complexity by 10x of handling all these formats.

We've recently been deploying LZ4, especially since the latest release [1.10.0 supports multi-threaded compression and decompression](https://github.com/lz4/lz4/releases/tag/v1.10.0)

Unfortunately [https://github.com/python-lz4/python-lz4](https://github.com/python-lz4/python-lz4) development seems to have ceased.

Is there any possibility for adding python-zstandard style library? or support for invoking "lz4" CLI for now?

**Comment by marcelm** [comment]
Thanks for the kind words!

My first thought is that adding LZ4 sounds reasonable. Are you interested in submitting a PR? A couple of thoughts in random order:
- Spawning `lz4` as a process would be easiest for now.
- Using python-lz4 as a library should also be needed, but the question is whether it should be a required or optional dependency. I don’t like that the lz4 wheel has 1.3 MB. This seems to be due to debugging info being included and could be reduced to around 100 kB. The Conda packages for lz4 itself plus the Python bindings (`lz4-c` and `lz4`) total <200 kB.
- python-lz4 apparently not being maintained is a bummer.
- Maybe the tests need some refactoring to avoid running too many of them?

@rhpvorderman What do you think?

**Comment by marcelm** [comment]
Ah, great to hear it’s maintained. I’d be happy to have lz4 support, but don’t have time myself to work on this at the moment. So the answer at the moment is: PRs welcome!

**Comment by rhpvorderman** [comment]
Yes, I am also in favour of adding LZ4. The python package not being ideal is not that bad as long as the conda package are good in terms of size. Those affect biocontainer sizes. 1.3MB is doable on most internet connections.
